### DISCUSSION TITLE
Splitting test/spec/builtins/diagnostics_spec.lua into smaller pieces

### DISCUSSION THREAD

**Discussion by eshepelyuk** [nvimtools/none-ls.nvim#25] [Ideas]
Hello @mochaaP 

wdyt about this idea ?
my main concern is that its rather hard to read at the moment. And the biggest concern is that there is no way to run particular test, but only entire file - and this also complicates test development.

**Comment by mochaaP** [comment]
It should be great, but be careful about the side effects of the mocked API, as some of them reset between tests, some don't. It's definitely welcomed to implement it! <3

**Reply by eshepelyuk** [reply]
my intention is to move out opacheck and regal linters tests to separate files \ spec. since i am currently doing some fix here and there.
